osu.Game/Database/RealmAccess.cs

```diff
@@ -986,6 +991,64 @@ private void applyMigrationsForVersion(Migration migration, ulong targetVersion)
 
                     break;
                 }
+
+                case 32:
+                {
+                    foreach (var score in migration.NewRealm.All<ScoreInfo>())
+                    {
+                        if (!score.IsLegacyScore || !score.Ruleset.IsLegacyRuleset())
+                            continue;
+
+                        string? replayFilename = score.Files.FirstOrDefault(f => f.Filename.EndsWith(@".osr", StringComparison.InvariantCultureIgnoreCase))?.File.GetStoragePath();
+                        if (replayFilename == null)
+                            continue;
+
+                        try
+                        {
+                            using (var stream = files.Store.GetStream(replayFilename))
```

Comment:
Can we move as much of this as possible to a helper method inside `StandardisedScoreMigrationTools`?

Reply:
Does https://github.com/ppy/osu/pull/24169/commits/9e4ffc8c12cdce5bb34fee06fafdec76ac41337e work for you?

Reply:
Seems a bit better yeah.